src/main/java/bral/fallingsand/SandComponent.java

```diff
@@ -0,0 +1,37 @@
+package bral.fallingsand;
+
+import javax.swing.*;
+import java.awt.*;
+
+public class SandComponent extends JComponent {
+
+    private final Sand sand;
+    private final int[][] currField;
+
+    public SandComponent(Sand sand) {
+        this.sand = sand;
+        this.currField = sand.getField();
+    }
+
+    @Override
+    protected void paintComponent(Graphics g) {
+        super.paintComponent(g);
+
+        g.translate(0, -getHeight());
```

Comment:
We translated the origin in the previous program because we wanted it in the lower left hand corner. 

Here we want the origin where it is, in the upper left hand corner.